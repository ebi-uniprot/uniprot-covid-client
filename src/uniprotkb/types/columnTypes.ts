export enum Column {
  threeD = '3d',
  absorption = 'absorption',
  accession = 'accession',
  ccAllergen = 'cc_allergen',
  ccAlternativeProducts = 'cc_alternative_products',
  ccBiotechnology = 'cc_biotechnology',
  ccCatalyticActivity = 'cc_catalytic_activity',
  ccCaution = 'cc_caution',
  ccCofactor = 'cc_cofactor',
  ccDevelopmentalStage = 'cc_developmental_stage',
  ccDisease = 'cc_disease',
  ccDisruptionPhenotype = 'cc_disruption_phenotype',
  ccDomain = 'cc_domain',
  ccActivityRegulation = 'cc_activity_regulation',
  ccFunction = 'cc_function',
  ccInduction = 'cc_induction',
  ccInteraction = 'cc_interaction',
  ccMassSpectrometry = 'cc_mass_spectrometry',
  ccMiscellaneous = 'cc_miscellaneous',
  ccPathway = 'cc_pathway',
  ccPharmaceutical = 'cc_pharmaceutical',
  ccPolymorphism = 'cc_polymorphism',
  ccPtm = 'cc_ptm',
  ccRnaEditing = 'cc_rna_editing',
  ccSequenceCaution = 'cc_sequence_caution',
  ccSimilarity = 'cc_similarity',
  ccSubcellularLocation = 'cc_subcellular_location',
  ccSubunit = 'cc_subunit',
  ccTissueSpecificity = 'cc_tissue_specificity',
  ccToxicDose = 'cc_toxic_dose',
  dateCreate = 'date_create',
  dateMod = 'date_mod',
  dateSeqMod = 'date_seq_mod',
  drAbcd = 'dr_abcd',
  drAllergome = 'dr_allergome',
  drArachnoserver = 'dr_arachnoserver',
  drAraport = 'dr_araport',
  drBgee = 'dr_bgee',
  drBindingdb = 'dr_bindingdb',
  drBiocyc = 'dr_biocyc',
  drBiogrid = 'dr_biogrid',
  drBiomuta = 'dr_biomuta',
  drBrenda = 'dr_brenda',
  drCarbonyldb = 'dr_carbonyldb',
  drCazy = 'dr_cazy',
  drCcds = 'dr_ccds',
  drCdd = 'dr_cdd',
  drCgd = 'dr_cgd',
  drChembl = 'dr_chembl',
  drChitars = 'dr_chitars',
  drCleanex = 'dr_cleanex',
  drCollectf = 'dr_collectf',
  drComplexportal = 'dr_complexportal',
  drCompluyeast2Dpage = 'dr_compluyeast-2dpage',
  drConoserver = 'dr_conoserver',
  drCorum = 'dr_corum',
  drCptac = 'dr_cptac',
  drCtd = 'dr_ctd',
  drDbsnp = 'dr_dbsnp',
  drDepod = 'dr_depod',
  drDictybase = 'dr_dictybase',
  drDip = 'dr_dip',
  drDisgenet = 'dr_disgenet',
  drDisprot = 'dr_disprot',
  drDmdm = 'dr_dmdm',
  drDnasu = 'dr_dnasu',
  drDosacCobs2Dpage = 'dr_dosac-cobs-2dpage',
  drDrugbank = 'dr_drugbank',
  drEchobase = 'dr_echobase',
  drEggnog = 'dr_eggnog',
  drElm = 'dr_elm',
  drEmbl = 'dr_embl',
  drEnsembl = 'dr_ensembl',
  drEnsemblbacteria = 'dr_ensemblbacteria',
  drEnsemblfungi = 'dr_ensemblfungi',
  drEnsemblmetazoa = 'dr_ensemblmetazoa',
  drEnsemblplants = 'dr_ensemblplants',
  drEnsemblprotists = 'dr_ensemblprotists',
  drEpd = 'dr_epd',
  drEsther = 'dr_esther',
  drEuhcvdb = 'dr_euhcvdb',
  drEupathdb = 'dr_eupathdb',
  drEvolutionarytrace = 'dr_evolutionarytrace',
  drExpressionatlas = 'dr_expressionatlas',
  drFlybase = 'dr_flybase',
  drGene3D = 'dr_gene3d',
  drGenecards = 'dr_genecards',
  drGenedb = 'dr_genedb',
  drGeneid = 'dr_geneid',
  drGenereviews = 'dr_genereviews',
  drGenetree = 'dr_genetree',
  drGenevisible = 'dr_genevisible',
  drGenewiki = 'dr_genewiki',
  drGenomernai = 'dr_genomernai',
  drGlyconnect = 'dr_glyconnect',
  drGo = 'dr_go',
  drGramene = 'dr_gramene',
  drGuidetopharmacology = 'dr_guidetopharmacology',
  drHamap = 'dr_hamap',
  drHgnc = 'dr_hgnc',
  drHogenom = 'dr_hogenom',
  drHpa = 'dr_hpa',
  drImgtGeneDb = 'dr_imgt_gene-db',
  drInparanoid = 'dr_inparanoid',
  drIntact = 'dr_intact',
  drInterpro = 'dr_interpro',
  drIptmnet = 'dr_iptmnet',
  drJpost = 'dr_jpost',
  drKegg = 'dr_kegg',
  drKo = 'dr_ko',
  drLegiolist = 'dr_legiolist',
  drLeproma = 'dr_leproma',
  drMaizegdb = 'dr_maizegdb',
  drMalacards = 'dr_malacards',
  drMassive = 'dr_massive',
  drMaxqb = 'dr_maxqb',
  drMerops = 'dr_merops',
  drMgi = 'dr_mgi',
  drMim = 'dr_mim',
  drMint = 'dr_mint',
  drMoondb = 'dr_moondb',
  drMoonprot = 'dr_moonprot',
  drMycoclap = 'dr_mycoclap',
  drNextprot = 'dr_nextprot',
  drNiagads = 'dr_niagads',
  drOgp = 'dr_ogp',
  drOma = 'dr_oma',
  drOpentargets = 'dr_opentargets',
  drOrphanet = 'dr_orphanet',
  drOrthodb = 'dr_orthodb',
  drPanther = 'dr_panther',
  drPatric = 'dr_patric',
  drPaxdb = 'dr_paxdb',
  drPdb = 'dr_pdb',
  drPdbsum = 'dr_pdbsum',
  drPeptideatlas = 'dr_peptideatlas',
  drPeroxibase = 'dr_peroxibase',
  drPfam = 'dr_pfam',
  drPharmgkb = 'dr_pharmgkb',
  drPhosphositeplus = 'dr_phosphositeplus',
  drPhylomedb = 'dr_phylomedb',
  drPir = 'dr_pir',
  drPirsf = 'dr_pirsf',
  drPlantreactome = 'dr_plantreactome',
  drPombase = 'dr_pombase',
  drPride = 'dr_pride',
  drPrints = 'dr_prints',
  drPro = 'dr_pro',
  drProdom = 'dr_prodom',
  drPromex = 'dr_promex',
  drProsite = 'dr_prosite',
  drProteomes = 'dr_proteomes',
  drProteomicsdb = 'dr_proteomicsdb',
  drPseudocap = 'dr_pseudocap',
  drReactome = 'dr_reactome',
  drRebase = 'dr_rebase',
  drRefseq = 'dr_refseq',
  drReproduction2Dpage = 'dr_reproduction-2dpage',
  drRgd = 'dr_rgd',
  drSabioRk = 'dr_sabio-rk',
  drSfld = 'dr_sfld',
  drSgd = 'dr_sgd',
  drSignalink = 'dr_signalink',
  drSignor = 'dr_signor',
  drSmart = 'dr_smart',
  drSmr = 'dr_smr',
  drString = 'dr_string',
  drSupfam = 'dr_supfam',
  drSwiss2Dpage = 'dr_swiss-2dpage',
  drSwisslipids = 'dr_swisslipids',
  drSwisspalm = 'dr_swisspalm',
  drTair = 'dr_tair',
  drTcdb = 'dr_tcdb',
  drTigrfams = 'dr_tigrfams',
  drTopdownproteomics = 'dr_topdownproteomics',
  drTreefam = 'dr_treefam',
  drTuberculist = 'dr_tuberculist',
  drUcd2Dpage = 'dr_ucd-2dpage',
  drUcsc = 'dr_ucsc',
  drUnicarbkb = 'dr_unicarbkb',
  drUnilectin = 'dr_unilectin',
  drUnipathway = 'dr_unipathway',
  drVectorbase = 'dr_vectorbase',
  drVgnc = 'dr_vgnc',
  drWbparasite = 'dr_wbparasite',
  drWorld2Dpage = 'dr_world-2dpage',
  drWormbase = 'dr_wormbase',
  drXenbase = 'dr_xenbase',
  drZfin = 'dr_zfin',
  ec = 'ec',
  errorGmodelPred = 'error_gmodel_pred',
  feature = 'feature',
  fragment = 'fragment',
  ftActSite = 'ft_act_site',
  ftBinding = 'ft_binding',
  ftCaBind = 'ft_ca_bind',
  ftCarbohyd = 'ft_carbohyd',
  ftChain = 'ft_chain',
  ftCoiled = 'ft_coiled',
  ftCompbias = 'ft_compbias',
  ftConflict = 'ft_conflict',
  ftCrosslnk = 'ft_crosslnk',
  ftDisulfide = 'ft_disulfide',
  ftDnaBind = 'ft_dna_bind',
  ftDomain = 'ft_domain',
  ftHelix = 'ft_helix',
  ftInitMet = 'ft_init_met',
  ftIntramem = 'ft_intramem',
  ftLipid = 'ft_lipid',
  ftMetal = 'ft_metal',
  ftModRes = 'ft_mod_res',
  ftMotif = 'ft_motif',
  ftMutagen = 'ft_mutagen',
  ftNonCon = 'ft_non_con',
  ftNonStd = 'ft_non_std',
  ftNonTer = 'ft_non_ter',
  ftNpBind = 'ft_np_bind',
  ftPeptide = 'ft_peptide',
  ftPropep = 'ft_propep',
  ftRegion = 'ft_region',
  ftRepeat = 'ft_repeat',
  ftSignal = 'ft_signal',
  ftSite = 'ft_site',
  ftStrand = 'ft_strand',
  ftTopDom = 'ft_top_dom',
  ftTransit = 'ft_transit',
  ftTransmem = 'ft_transmem',
  ftTurn = 'ft_turn',
  ftUnsure = 'ft_unsure',
  ftVarSeq = 'ft_var_seq',
  ftVariant = 'ft_variant',
  ftZnFing = 'ft_zn_fing',
  organelle = 'organelle',
  geneNames = 'gene_names',
  geneOln = 'gene_oln',
  geneOrf = 'gene_orf',
  genePrimary = 'gene_primary',
  geneSynonym = 'gene_synonym',
  go = 'go',
  goC = 'go_c',
  goF = 'go_f',
  goId = 'go_id',
  goP = 'go_p',
  id = 'id',
  keyword = 'keyword',
  keywordid = 'keywordid',
  kinetics = 'kinetics',
  length = 'length',
  lineage = 'lineage',
  mappedPmId = 'mapped_pm_id',
  mass = 'mass',
  matchedText = 'matched_text',
  mnemonic = 'mnemonic',
  organism = 'organism',
  organismHost = 'organism_host',
  organismId = 'organism_id',
  phDependence = 'ph_dependence',
  pmId = 'pm_id',
  proteinExistence = 'protein_existence',
  proteinFamilies = 'family',
  proteinName = 'protein_name',
  redoxPotential = 'redox_potential',
  reviewed = 'reviewed',
  score = 'score',
  sequence = 'sequence',
  sequenceVersion = 'sequence_version',
  taxId = 'tax_id',
  tempDependence = 'temp_dependence',
  tlAll = 'tl_all',
  tlClass = 'tl_class',
  tlCohort = 'tl_cohort',
  tlFamily = 'tl_family',
  tlForma = 'tl_forma',
  tlGenus = 'tl_genus',
  tlInfraclass = 'tl_infraclass',
  tlInfraorder = 'tl_infraorder',
  tlKingdom = 'tl_kingdom',
  tlOrder = 'tl_order',
  tlParvorder = 'tl_parvorder',
  tlPhylum = 'tl_phylum',
  tlSpecies = 'tl_species',
  tlSpeciesGroup = 'tl_species_group',
  tlSpeciesSubgroup = 'tl_species_subgroup',
  tlSubclass = 'tl_subclass',
  tlSubcohort = 'tl_subcohort',
  tlSubfamily = 'tl_subfamily',
  tlSubgenus = 'tl_subgenus',
  tlSubkingdom = 'tl_subkingdom',
  tlSuborder = 'tl_suborder',
  tlSubphylum = 'tl_subphylum',
  tlSubspecies = 'tl_subspecies',
  tlSubtribe = 'tl_subtribe',
  tlSuperclass = 'tl_superclass',
  tlSuperfamily = 'tl_superfamily',
  tlSuperkingdom = 'tl_superkingdom',
  tlSuperorder = 'tl_superorder',
  tlSuperphylum = 'tl_superphylum',
  tlTribe = 'tl_tribe',
  tlVarietas = 'tl_varietas',
  tools = 'tools',
  uniparcId = 'uniparc_id',
  version = 'version',
}

export type SortableColumn =
  | Column.accession
  | Column.score
  | Column.geneNames
  | Column.length
  | Column.mass
  | Column.mnemonic
  | Column.proteinName
  | Column.organism;
