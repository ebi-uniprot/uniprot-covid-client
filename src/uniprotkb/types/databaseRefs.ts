export enum DatabaseCategory {
  CHEMISTRY = 'CHEMISTRY',
  DOMAIN = 'FMD',
  EXPRESSION = 'GEP',
  FAMILY = 'PFAM',
  GEL = '2DG',
  GENOME = 'GMA',
  INTERACTION = 'PPI',
  ORGANISM = 'ORG',
  OTHER = 'OTHER',
  PATHWAY = 'EAP',
  PHYLOGENOMIC = 'PLG',
  POLYMORPHISM = 'PMD',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PAM',
  PTM = 'PTM',
  SEQUENCE = 'SEQ',
  STRUCTURE = '3DS',
}

export type AttributesItem = {
  name: string;
  xmlTag: string;
  uriLink?: string;
};

export type DatabaseInfoPoint = {
  name: string;
  displayName: string;
  category: string;
  uriLink: string;
  attributes?: AttributesItem[];
  implicit?: boolean;
};

export type DatabaseInfo = DatabaseInfoPoint[];
