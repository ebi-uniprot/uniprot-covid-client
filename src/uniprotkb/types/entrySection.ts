enum EntrySection {
  Function = 'Function',
  FamilyAndDomains = 'Family & Domains',
  Expression = 'Expression',
  Interaction = 'Interaction',
  // Miscellaneous = 'Miscellaneous',
  NamesAndTaxonomy = 'Names & Taxonomy',
  PathologyAndBioTech = 'Pathology & BioTech',
  ProteinProcessing = 'Protein Processing',
  Sequence = 'Sequence',
  Structure = 'Structure',
  SubCellularLocation = 'Subcellular Location',
  ExternalLinks = 'External Links',
}

export default EntrySection;
