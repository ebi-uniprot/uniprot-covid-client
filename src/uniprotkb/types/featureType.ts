enum FeatureType {
  INIT_MET = 'initiator methionine',
  SIGNAL = 'signal peptide',
  PROPEP = 'propeptide',
  TRANSIT = 'transit peptide',
  CHAIN = 'chain',
  PEPTIDE = 'peptide',
  TOPO_DOM = 'topological domain',
  TRANSMEM = 'transmembrane region',
  INTRAMEM = 'intramembrane region',
  DOMAIN = 'Other domain of interest',
  REPEAT = 'repeat',
  CA_BIND = 'calcium-binding region',
  ZN_FING = 'zinc finger region',
  DNA_BIND = 'DNA-binding region',
  NP_BINDL = 'Nucleotide-binding region',
  REGION = 'region of interest',
  COILED = 'coiled-coil region',
  MOTIF = 'short sequence motif',
  COMPBIAS = 'compositionally biased region',
  ACT_SITE = 'active site',
  METAL = 'metal ion-binding site',
  BINDING = 'Other binding site',
  SITE = 'Other site of interest',
  NON_STD = 'non-standard amino acid',
  MOD_RES = 'Post-translationally modified residue',
  LIPID = 'lipid moiety-binding region',
  CARBOHYD = 'glycosylation site',
  DISULFID = 'disulfide bond',
  CROSSLNK = 'cross-link',
  VAR_SEQ = 'splice variant',
  VARIANT = 'Sequence variation',
  MUTAGEN = 'mutagenesis site',
  UNSURE = 'Uncertainty in sequence',
  CONFLICT = 'sequence conflict',
  NON_CONS = 'non-consecutive residues',
  NON_TER = 'non-terminal residue',
  HELIX = 'helix',
  TURN = 'turn',
  STRAND = 'strand',
}

export default FeatureType;
