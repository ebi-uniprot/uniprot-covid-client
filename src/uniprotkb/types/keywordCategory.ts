export enum KeywordCategory {
  BIOLOGICAL_PROCESS = 'Biological process',
  CELLULAR_COMPONENT = 'Cellular component',
  CODING_SEQUENCE_DIVERSITY = 'Coding sequence diversity',
  DEVELOPMENTAL_STAGE = 'Developmental stage',
  DISEASE = 'Disease',
  DOMAIN = 'Domain',
  LIGAND = 'Ligand',
  MOLECULAR_FUNCTION = 'Molecular function',
  PTM = 'PTM',
  TECHNICAL_TERM = 'Technical term',
}

export default KeywordCategory;
