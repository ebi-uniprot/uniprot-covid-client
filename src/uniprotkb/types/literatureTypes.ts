import { Xref } from './commentTypes';

export type Citation = {
  citationType?: string;
  authors?: string[];
  citationCrossReferences?: Xref[];
  title?: string;
  publicationDate?: number;
  journal?: string;
  firstPage?: number;
  lastPage?: number;
  volume?: number;
  completeAuthorList?: boolean;
  literatureAbstract?: string;
  authoringGroup?: string[];
  submissionDatabase?: string;
};

export type Reference = {
  citation: Citation;
  referencePositions?: string[];
  referenceComments?: {
    value: string;
    type: string;
  }[];
};

export type LiteratureStatistics = {
  reviewedProteinCount?: number;
  unreviewedProteinCount?: number;
  mappedProteinCount?: number;
};

export type LiteratureForProteinAPI = {
  reference: Reference;
  statistics?: LiteratureStatistics;
  categories?: string[];
  publicationSource?: string;
};

export type LiteratureAPI = {
  citation: Citation;
  statistics?: LiteratureStatistics;
};
