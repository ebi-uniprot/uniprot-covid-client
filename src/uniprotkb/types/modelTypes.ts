export type Evidence = {
  evidenceCode: string; // TODO this needs to be an ECO instead of string
  source?: string;
  id?: string;
};

export type ValueWithEvidence = {
  value: string;
  evidences: Evidence[];
};

export enum PropertyKey {
  AllergenName = 'AllergenName',
  Chains = 'Chains',
  Component = 'Component',
  Description = 'Description',
  Disease = 'Disease',
  EntryName = 'EntryName',
  ExpressionPatterns = 'ExpressionPatterns',
  FamilyName = 'FamilyName',
  Fingerprint = 'Fingerprint',
  GeneDesignation = 'GeneDesignation',
  GeneId = 'GeneId',
  GeneName = 'GeneName',
  GenericName = 'GenericName',
  GoEvidenceType = 'GoEvidenceType',
  GoTerm = 'GoTerm',
  Interactions = 'Interactions',
  MatchStatus = 'MatchStatus',
  Method = 'Method',
  MoleculeType = 'MoleculeType',
  NucleotideSequenceId = 'NucleotideSequenceId',
  OrganismId = 'OrganismId',
  OrganismName = 'OrganismName',
  PathwayName = 'PathwayName',
  Project = 'Project',
  ProteinId = 'ProteinId',
  RectionId = 'RectionId',
  Resolution = 'Resolution',
  Status = 'Status',
  ToxinName = 'ToxinName',
  ToxonomicScope = 'ToxonomicScope',
  Type = 'Type',
}

export type Property = {
  key?: PropertyKey;
  value?: string;
};
