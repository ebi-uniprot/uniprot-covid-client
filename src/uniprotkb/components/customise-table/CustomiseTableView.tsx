import React from 'react';
import ColumnSelectContainer from '../column-select/ColumnSelectContainer';
import { Column } from '../../types/columnTypes';
import './styles/customise-table.scss';

type CustomiseTableViewProps = {
  selectedColumns: Column[];
  onChange: (columndIds: Column[]) => void;
  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  onCancel: () => void;
};

const CustomiseTableView: React.FC<CustomiseTableViewProps> = ({
  selectedColumns,
  onChange,
  onSubmit,
  onCancel,
}) => (
  <form
    onSubmit={onSubmit}
    className="customise-table"
    data-testid="customise-table-form"
  >
    <ColumnSelectContainer
      onChange={onChange}
      selectedColumns={selectedColumns}
    />
    <div className="button-group customise-table--cancel-submit-buttons">
      <button
        className="button secondary"
        type="button"
        onClick={onCancel}
        data-testid="customise-table-cancel-button"
      >
        Cancel
      </button>
      <button className="button" type="submit">
        Save
      </button>
    </div>
  </form>
);

export default CustomiseTableView;
