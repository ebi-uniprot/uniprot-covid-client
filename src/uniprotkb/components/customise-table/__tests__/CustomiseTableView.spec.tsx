import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderWithRedux from '../../../../shared/__test-helpers__/RenderWithRedux';
import initialState from '../../../../app/state/rootInitialState';
import CustomiseTableView from '../CustomiseTableView';
import { Column } from '../../../types/columnTypes';

describe('CustomiseTableView component', () => {
  let props, renderedWithRedux;
  beforeEach(() => {
    props = {
      selectedColumns: [Column.accession, Column.ccAllergen],
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      onCancel: jest.fn(),
    };
    renderedWithRedux = renderWithRedux(<CustomiseTableView {...props} />, {
      initialState,
      location: '/customise-table',
    });
  });

  test('should render', () => {
    const { asFragment } = renderedWithRedux;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call onSubmit when submit button is clicked', () => {
    const { getByTestId } = renderedWithRedux;
    const form = getByTestId('customise-table-form');
    fireEvent.submit(form);
    expect(props.onSubmit).toHaveBeenCalled();
  });

  test('should call onCancel when cancel button is clicked', () => {
    const { getByTestId } = renderedWithRedux;
    const button = getByTestId('customise-table-cancel-button');
    fireEvent.click(button);
    expect(props.onCancel).toHaveBeenCalled();
  });
});
