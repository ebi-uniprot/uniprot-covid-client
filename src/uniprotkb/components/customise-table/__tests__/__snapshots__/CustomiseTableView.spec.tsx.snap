// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomiseTableView component should render 1`] = `
<DocumentFragment>
  <form
    class="customise-table"
    data-testid="customise-table-form"
  >
    <div
      class="loader-container"
    >
      <svg
        class="loader"
        height="100"
        viewBox="0 0 100 100"
        width="100"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M9 50c6.833-10.667 13.667-16 20.5-16S43.167 39.333 50 50s13.667 16 20.5 16S84.167 60.667 91 50"
            stroke="#3ca3ca"
            stroke-linecap="square"
            stroke-width="2"
          />
          <circle
            cx="9"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="30"
            cy="34"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="50"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="70"
            cy="66"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="91"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
        </g>
      </svg>
    </div>
    <div
      class="button-group customise-table--cancel-submit-buttons"
    >
      <button
        class="button secondary"
        data-testid="customise-table-cancel-button"
        type="button"
      >
        Cancel
      </button>
      <button
        class="button"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</DocumentFragment>
`;
